### ISSUE TITLE
undo incorrectly snapping to stuff

### ISSUE THREAD

**Issue by maryfries** [jmoenig/Snap--Build-Your-Own-Blocks#1715]
I just made a mistake and used undo, and Snap put the script back where it was before _and attached it to something it had not been attached to_.

**Comment by maryfries** [comment]
Here's another thing I've seen happen with undo...

![image](https://cloud.githubusercontent.com/assets/6549787/25047695/5ed58036-2106-11e7-91c0-0a37d37f914d.png)

**Comment by jmoenig** [comment]
yes, my current "undo" has quirks, especially with reporters. The next version from @brollb will be much better!

**Comment by maryfries** [comment]
More buggy results of undo:

![image](https://user-images.githubusercontent.com/6549787/40386591-279758a0-5dd8-11e8-926b-c1d5d943d964.png)
